### ISSUE TITLE
declare / typeset / etc. don't validate flags when statically parsed

### ISSUE THREAD

**Issue by andychu** [oilshell/oil#391]
reported by @lhoursquentin on Zulip

```
I noticed declare/typeset/local/readonly (weirdly not export) accept anything starting with dashes:

$ osh -c 'declare --hello-- a=1; echo "$a"'
1
```

**Comment by andychu** [comment]
Similar to #343

**Comment by andychu** [comment]
Released with 0.7.pre2: https://www.oilshell.org/release/0.7.pre2/
